tools/gen-ts-glue.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * Mixing glue with lint is probably something you never want to do in real
+ * life, but in this project it helps the linter do its job. And sometimes the
+ * test suite too.
+ *
+ * This script creates `index.ts` files in each package directory that simply
+ * re-export the exports of the real main file. This is necessary for when all
+ * the packages are run under a common context (instead of each individually).
+ * This is because the symlinks bolt makes for each package are descendants of
```

Comment:
Interesting. So in theory when we move away from bolt in the future this would be unnecessary?

Reply:
I think other tools will still have this issue. The issue stems from when we run a command on "all packages" (run from the project root and have it traverse into everything including the packages), which is in contrast to running a command "per-package" (running the command in every package with that package as the root).

This difference is significant because if you start in `root/` and traverse down to `root/packages/my-package/node_modules/symlink-to-my-other-package` (a symlink) and that takes you to `root/packages/my-other-package`, you're still under the root context `root/` which means that tooling will *not* look for a `package.json` file because it does not think you're checking a dependency. That is why the index file solution works, because the index file *will* be automatically checked when the dependency resolution thinks it's still under the root context (which is true because we chose the root to be the project, rather than a specific package).

Does that make more sense?